_Group Projects/Group_5/Week8_Assignment/Group5_Week8_Simulation.do

```diff
@@ -0,0 +1,131 @@
+
+*Public quality
+gen public_quality = rnormal(100,15)  // Arbitrary score assessing public quality with mean of 100 and SD of 15
+la var public_quality "Quality score of public school"
+
+*Charter quality
+gen charter_quality = rnormal(115,15) // Arbitrary score assessing charter quality, where mean is one SD above that of public quality distribution; we make this assumption because our literature review shows charters tend to produce higher test scores
+la var charter_quality "Quality score of charter school"
+
+*Applicants
+gen applicants = .
+forval i = 1/5000 { // Loop through each row
+	count if charter_id==`i' // Count the number of kids applying to charter in current row
+	replace applicants = r(N) if charter_id==`i' // Replace applicants with the number of kids applying to same charter
+}
+la var applicants "Total applicants to same charter"
+
+*Oversubscribed
+gen oversubscribed = . // Charter is oversubscribed if applicants exceeds school capacity
+replace oversubscribed = 1 if applicants>capacity
+replace oversubscribed = 0 if applicants<=capacity
+la var oversubscribed "=1 if applicants > capacity"
+
+*Random lottery mechanism
+gen tempvar = .
+replace tempvar = rnormal() if oversubscribed==1 // Randomly assign number to each kid who applied to oversubscribed school
+gen lottery_rank = .
+la var lottery_rank "Random lottery ranking for oversubscribed charters"
+bysort charter_id (tempvar): replace lottery_rank = _n // Sort tempvar in ascending order for applicants to oversubscribed schools, and then fill in lottery rank
+drop tempvar // Temporary variable no longer needed
+
+*Admitted to charter
+gen admitted = .
+replace admitted = 1 if lottery_rank<=capacity // =1 when child's lottery rank is less than school capacity
+replace admitted = 1 if oversubscribed==0 // =1 when no lottery occured
+replace admitted = 0 if oversubscribed==1 & lottery_rank>capacity // =0 for lottery losers
+la var admitted "=1 if child was admitted to charter"
+
+*Test score at charter school
+gen ts_charter = .
+replace ts_charter = iq^0.5 * charter_quality^0.5 if admitted==1 // Cobb-Douglas education production function, where a = 0.5; we chose the number based on the education production function used by Polachek, Kniesner, and Harwood (1978) in the Journal of Educational Statistics
+la var ts_charter "Child's test score if he/she attends charter"
+
+*Test score at public school
+gen ts_public = .
+replace ts_public = iq^0.5 * public_quality^0.5 // Cobb-Douglas education production function, where a = 0.5; see justification for number above
+la var ts_public "Child's test score if he/she attends public school"
+
+*School attended
+gen attend = .
+replace attend = 1 if admitted==1 & ts_charter>=ts_public // =1 if child was admitted to charter and would achieve higher test score attending charter; recall hypothetical test score at charter was generated above using Cobb-Douglas education production function
```

Comment:
Do you replace children who reject the offer with other children? What happens to empty slots at charter schools? How many are empty? I would check this!